cmd/collectors/zapi/plugins/volume/volume.go

```diff
@@ -20,84 +20,171 @@ type Volume struct {
 }
 
 type aggrData struct {
-	aggrUUID string
-	aggrName string
+	uuid string
+	name string
+}
+
+type volumeClone struct {
+	name           string
+	svm            string
+	parentSnapshot string
+	parentVolume   string
+	parentSvm      string
+	splitEstimate  string
 }
 
 func New(p *plugin.AbstractPlugin) plugin.Plugin {
 	return &Volume{AbstractPlugin: p}
 }
 
-func (my *Volume) Init() error {
+func (v *Volume) Init() error {
 
 	var err error
 
-	if err = my.InitAbc(); err != nil {
+	if err = v.InitAbc(); err != nil {
 		return err
 	}
 
-	if my.client, err = zapi.New(conf.ZapiPoller(my.ParentParams), my.Auth); err != nil {
-		my.Logger.Error().Stack().Err(err).Msg("connecting")
+	if v.client, err = zapi.New(conf.ZapiPoller(v.ParentParams), v.Auth); err != nil {
+		v.Logger.Error().Stack().Err(err).Msg("connecting")
 		return err
 	}
 
-	if err = my.client.Init(5); err != nil {
+	if err = v.client.Init(5); err != nil {
 		return err
 	}
 
-	my.aggrsMap = make(map[string]string)
+	v.aggrsMap = make(map[string]string)
 
 	// Assigned the value to currentVal so that plugin would be invoked first time to populate cache.
-	my.currentVal = my.SetPluginInterval()
+	v.currentVal = v.SetPluginInterval()
 
 	return nil
 }
 
-func (my *Volume) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, error) {
+func (v *Volume) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, error) {
 
-	data := dataMap[my.Object]
-	if my.currentVal >= my.PluginInvocationRate {
-		my.currentVal = 0
+	data := dataMap[v.Object]
+	if v.currentVal >= v.PluginInvocationRate {
+		v.currentVal = 0
 
 		// invoke disk-encrypt-get-iter zapi and populate disk info
-		disks, err1 := my.getEncryptedDisks()
+		disks, err1 := v.getEncryptedDisks()
 		// invoke aggr-status-get-iter zapi and populate aggr disk mapping info
-		aggrDiskMap, err2 := my.getAggrDiskMapping()
+		aggrDiskMap, err2 := v.getAggrDiskMapping()
 
 		if err1 != nil {
 			if errors.Is(err1, errs.ErrNoInstance) {
-				my.Logger.Debug().Err(err1).Msg("Failed to collect disk data")
+				v.Logger.Debug().Err(err1).Msg("Failed to collect disk data")
 			} else {
-				my.Logger.Error().Err(err1).Msg("Failed to collect disk data")
+				v.Logger.Error().Err(err1).Msg("Failed to collect disk data")
 			}
 		}
 		if err2 != nil {
 			if errors.Is(err2, errs.ErrNoInstance) {
-				my.Logger.Debug().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
+				v.Logger.Debug().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
 			} else {
-				my.Logger.Error().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
+				v.Logger.Error().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
 			}
 		}
 		// update aggrsMap based on disk data and addr disk mapping
-		my.updateAggrMap(disks, aggrDiskMap)
+		v.updateAggrMap(disks, aggrDiskMap)
+	}
+
+	volumeCloneMap, err := v.getVolumeCloneInfo()
```

Comment:
Do we want to call this zapi at each poll or as other zapi call happens at plugin frequency to update map but still update volume labels at each poll from cache map ?